camelot/core/item_model/query_proxy.py

```diff
@@ -107,7 +107,8 @@ def index(self, obj):
             return self._indexed_objects[obj]
         except KeyError:
             i = self._objects.index(obj)
-            self._indexed_objects[i+self._length] = obj
+            length = self._length if self._length is not None else len(self)
+            self._indexed_objects[i+length] = obj
```

Comment:
?? deze case moeten we denk ik expliciet unit testen, nl 'index' callen terwijl _length nog nt gekend is

als ik de code lees, dan denk ik dat 'i+length' eentje te ver  gaat zijn, omdat lenght de som is van _length + len(_objects)

Reply:
Zonder deze aanpassing kreeg ik volgende exception bij het uitvoeren van de test_add_new_object test:

```
Traceback (most recent call last):
  File "/home/timvdm/vortex-blue-launcher/vfinance/src/camelot/core/item_model/query_proxy.py", line 107, in index
    return self._indexed_objects[obj]
KeyError: <vfinance.model.bank.natuurlijke_persoon.NatuurlijkePersoon object at 0x7f6d939969a0>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/timvdm/vortex-blue-launcher/vfinance/src/camelot/view/model_thread/signal_slot_model_thread.py", line 71, in execute
    result = self._request( *self._args )
  File "/home/timvdm/vortex-blue-launcher/vfinance/src/camelot/view/action_runner.py", line 136, in _iterate_until_blocking
    result = generator_method( *args )
  File "/home/timvdm/vortex-blue-launcher/vfinance/src/camelot/view/crud_action.py", line 283, in model_run
    row = model_context.proxy.index(obj)
  File "/home/timvdm/vortex-blue-launcher/vfinance/src/camelot/core/item_model/query_proxy.py", line 112, in index
    self._indexed_objects[i+length] = obj
TypeError: unsupported operand type(s) for +: 'int' and 'NoneType'
```

Ik heb nu een RowCount action step toegevoegd aan die test en dan is deze aanpassing niet meer nodig.